ironfish/src/rpc/routes/eth/getBlockByNumber.ts

```diff
@@ -0,0 +1,131 @@
+  logsBloom: string | null
+  miner: string
+  mixHash: string
+  nonce: string | null
+  number: string | null
+  parentHash: string
+  receiptsRoot: string
+  sha3Uncles: string
+  size: string
+  stateRoot: string
+  timestamp: string
+  totalDifficulty: string
+  transactions: (string | EthRpcTransaction)[]
+  transactionsRoot: string
+  uncles: string[]
+}
+
+export const GetBlockByNumberResponseSchema: yup.ObjectSchema<GetBlockByNumberResponse> = yup
+  .object({
+    baseFeePerGas: yup.string().notRequired(),
+    difficulty: yup.string().defined(),
+    extraData: yup.string().defined(),
+    gasLimit: yup.string().defined(),
+    gasUsed: yup.string().defined(),
+    hash: yup.string().nullable().defined(),
+    logsBloom: yup.string().nullable().defined(),
+    miner: yup.string().defined(),
+    mixHash: yup.string().defined(),
+    nonce: yup.string().nullable().defined(),
+    number: yup.string().nullable().defined(),
+    parentHash: yup.string().defined(),
+    receiptsRoot: yup.string().defined(),
+    sha3Uncles: yup.string().defined(),
+    size: yup.string().defined(),
+    stateRoot: yup.string().defined(),
+    timestamp: yup.string().defined(),
+    totalDifficulty: yup.string().defined(),
+    transactions: yup.array<string | EthRpcTransaction>().defined(),
+    transactionsRoot: yup.string().defined(),
+    uncles: yup.array().of(yup.string().defined()).defined(),
+  })
+  .defined()
+
+registerEthRoute<typeof GetBlockByNumberRequestSchema, GetBlockByNumberResponse>(
+  `eth_getBlockByNumber`,
+  `${ApiNamespace.eth}/getBlockByNumber`,
+  GetBlockByNumberRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isInstanceOf(node, FullNode)
+
+    const [blockNumber, transactionDetailFlag] = request.data
+
+    const block: Block | null = await node.chain.getBlockAtSequence(parseInt(blockNumber))
+    if (!block) {
+      throw new RpcNotFoundError(`Block ${blockNumber} not found`)
+    }
+
+    const blockHeader = block.header
+    const transactions: (string | EthRpcTransaction)[] = []
+    let index = 0
+    for (const transaction of block.transactions) {
+      if (transaction.evm) {
+        const txInfo = transactionDetailFlag
+          ? blockTransactionToEthRpcTransaction(transaction, block.header, index)
+          : EthUtils.bytesToHex(evmDescriptionToLegacyTransaction(transaction.evm).hash())
+        transactions.push(txInfo)
+      }
+      index += 1
+    }
+
+    // TODO handle mocked fields
+    const response: GetBlockByNumberResponse = {
+      baseFeePerGas: '0x0',
+      difficulty: '0x0',
+      extraData: '0x',
+      gasLimit: '0x0',
+      gasUsed: '0x0',
+      hash: EthUtils.prefix0x(blockHeader.hash.toString('hex')),
+      logsBloom: '0x',
+      miner: '0x',
+      mixHash: '0x',
+      nonce: '0x',
+      number: EthUtils.prefix0x(blockHeader.sequence.toString(16)),
+      parentHash: EthUtils.prefix0x(blockHeader.previousBlockHash.toString('hex')),
+      receiptsRoot: '0x',
```

Comment:
@hughy is this root available or the `transactionsRoot`? I think not, just verifying.

Reply:
receipts root isn't available right now, but we could create the tree and root for receipts on a block if we need to. I think `transactionsRoot` should be the same as `transactionCommitment` from the block header